### ISSUE TITLE
Responsive Widget's children do not resize with the parent

### ISSUE THREAD

**Issue by dearkarm** [iced-rs/iced#1202]
Children are assigned the correct size when the Widget is first created, but are not updated later unless Size is passed to them explicitly (and they have built in functionality to use it).

Example:
```
use iced::{Container, Element, Length, Sandbox, Settings, Text};
use iced::alignment::Horizontal;
use iced_lazy::Responsive;
use iced_native::alignment::Vertical;

fn main() -> iced::Result {
    TestApp::run(Settings::default())
}

struct TestApp {
    responsive: iced_lazy::responsive::State,
}

impl Sandbox for TestApp {
    type Message = ();

    fn new() -> Self {
        Self {
            responsive: Default::default()
        }
    }

    fn title(&self) -> String {
        "Responsive test".to_string()
    }

    fn update(&mut self, _message: Self::Message) {}

    fn view(&mut self) -> Element<'_, Self::Message> {
        Responsive::new(&mut self.responsive, |size| {
            Container::new(
                Text::new("Test".to_string())
            )
                .width(Length::Fill)
                .height(Length::Fill)
                .align_x(Horizontal::Center)
                .align_y(Vertical::Center)
                .into()
        })
            .into()
    }
}
```

`Cargo.toml`

```
[package]
name = "responsive_test"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
iced = { git = "https://github.com/iced-rs/iced.git", branch = "master", features=["tokio"]}
iced_native = { git = "https://github.com/iced-rs/iced.git", branch = "master"}
iced_lazy = { git = "https://github.com/iced-rs/iced.git", branch = "master"}
```

This is the main window after the program is ran:
![Screenshot 2022-01-19 at 22 07 38](https://user-images.githubusercontent.com/22744577/150214056-ef8b3c1e-4a8f-4e03-a719-0df87ee35db4.png)

The same window after resizing:
![Screenshot 2022-01-19 at 22 07 48](https://user-images.githubusercontent.com/22744577/150214106-141515f4-d499-4c37-a7b3-e14cfa3aad7d.png)

Adding this to the Container:
```
                .max_height(size.height as u32)
                .max_width(size.width as u32)
```

Causes resizing, however it does not look smooth. I tried this approach with my own widgets and I was getting incorrect sizing, i.e. during Window resize there'd be white strips indicating the Widget does not occupy the entire available space.

**Comment by hecrj** [comment]
Good catch! Should be fixed by a3744ec97a462557ab9a3d2509f02263f425fd28.

**Comment by dearkarm** [comment]
Works great. Thanks for the quick fix, Hector :)
